src/App.tsx

```diff
@@ -1,223 +1,147 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  memo,
+  useCallback,
+  useContext, useEffect, useMemo, useState,
+} from 'react';
+import {
+  getTodos,
+  creatTodo,
+  deletingTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
-
-export const App: React.FC = () => {
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+// eslint-disable-next-line import/no-cycle
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo, TodoCompleteStatus } from './types/Todo';
+import { todoFilteredCompleted, getTodoCompletedId } from './helpers/helpers';
+
+export const App: React.FC = memo(() => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoFilterComplete,
+    setTodoFilterComplete] = useState(TodoCompleteStatus.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds]
+    = useState<number[]>([]);
+
+  const showErrorMessage = useCallback((error: string) => {
+    setErrorMessage(error);
+
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(todo => (
+          setTodos(todo)))
+        .catch(() => showErrorMessage('Failed request for todos'));
     }
   }, []);
 
+  const userId = user ? user.id : 0;
+
+  const addTodo = async (todoData: Omit<Todo, 'id'>) => {
+    try {
+      const tempTodo = {
+        ...todoData,
+        id: 0,
+      };
+
+      setTemporaryTodo(tempTodo);
+
+      const newTodo = await creatTodo(todoData);
+
+      setTodos(currentTodoArray => [...currentTodoArray, newTodo]);
+      setTemporaryTodo(null);
```

Comment:
Use `finally` keyword, so you don't duplicate code inside `try` and `catch` instructions.
Example:
```
try {
} catch {
} finally {
 // delete temporary todo
}